### PR TITLE
Move memchr behind an optional feature

### LINKED ISSUE #76: msrv 1.48 broken
https://github.com/smol-rs/futures-lite/actions/runs/6060417025/job/16444572020

### PR DESCRIPTION
**Pull request by notgull**
Closes #76

### CODE REVIEW

**Comment by taiki-e** [comment]
You have to add memchr feature to the value of --skip.
https://github.com/smol-rs/futures-lite/blob/32cc46671c9076753a1ea0ad278695ac4f462199/.github/workflows/ci.yml#L45

File: Cargo.toml [hunk 1]
```diff
@@ -21,15 +21,16 @@ exclude = ["/.*"]
 
 [features]
 default = ["race", "std"]
-std = ["alloc", "fastrand/std", "futures-io", "parking", "memchr"]
+std = ["alloc", "fastrand/std", "futures-io", "parking"]
 alloc = []
+memchr = ["std", "memchr_crate"]
```
Comment (taiki-e) [inline Cargo.toml:26]: Considering the future support of core::io, I wonder if the memchr feature should not enable the std feature. However, I'm not sure if that will happen during the period when 2.x is the latest version.

**Comment by taiki-e** [comment]
LGTM

File: Cargo.toml (hunk 1 above)
Comment (notgull) [reply Cargo.toml:26]: As of now `futures-lite` uses `futures-io` v0.3. I expect that v0.3 probably won't export `core::io` traits without a breaking change, right? So I think we are secure in doing this for now.

Comment (taiki-e) [reply Cargo.toml:26]: futures-io has (enabled by default) `std` feature for future compatibility about this.

Comment (notgull) [reply Cargo.toml]: Good point, I've decouple `std` and `memchr`.

**Review by taiki-e — APPROVED** [review]
Thanks!

**Pull request merged**